lnwallet/btcwallet/btcwallet.go

```diff
@@ -1199,33 +1200,98 @@ func (b *BtcWallet) ListUnspentWitness(minConfs, maxConfs int32,
 // already published to the network (either in the mempool or chain) no error
 // will be returned.
 func (b *BtcWallet) PublishTransaction(tx *wire.MsgTx, label string) error {
-	if err := b.wallet.PublishTransaction(tx, label); err != nil {
+	// handleErr is a helper closure that handles the error from
+	// PublishTransaction and TestMempoolAccept.
+	handleErr := func(err error) error {
+		var (
+			doubleSpendErr      *base.ErrDoubleSpend
+			replacementErr      *base.ErrReplacement
+			mempoolFeeErr       *base.ErrMempoolFee
+			inMempoolErr        *base.ErrInMempool
+			alreadyConfirmedErr *base.ErrAlreadyConfirmed
+		)
+
 		// If we failed to publish the transaction, check whether we
 		// got an error of known type.
-		switch err.(type) {
+		switch {
 		// If the wallet reports a double spend, convert it to our
 		// internal ErrDoubleSpend and return.
-		case *base.ErrDoubleSpend:
+		case errors.As(err, &doubleSpendErr):
```

Comment:
👍

Reply:
Kinda annoying how we need to declare it first before we're able to use `errors.As` I think we could do something like this maybe: 
```go
func ErrorAs[E error](targetErr error) bool {
    var err E
    return errors.As(targetErr, &err)
}
```

So we use type params to reduce that duplication above where we need to declare the var so we can pass in the pointer. 

You'd call it like: 
```go
if ErrorAs[base.ErrInMempool(err) {
}
```

Non-blocking, just something that's bothered me a bit in the past re the API, and here's a straight forward way we can improve it a bit. You can also have it return the error as well, then a nil check is the same as the bool check.

Reply:
Very nice! Updated the `lnutils` to have this method as I too find it annoying, thanks!